From fb6cf510e80a644187c2ddf498763696bb4d9d82 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sat, 1 Nov 2025 13:58:26 -0400
Subject: [PATCH 27/28] Patch out zlib.h headers

---
 Modules/binascii.c   | 10 +++++-----
 Modules/zlibmodule.c |  2 +-
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/Modules/binascii.c b/Modules/binascii.c
index 6bb01d1..73cd97c 100644
--- a/Modules/binascii.c
+++ b/Modules/binascii.c
@@ -61,7 +61,7 @@
 #include "pycore_long.h"          // _PyLong_DigitValue
 #include "pycore_strhex.h"        // _Py_strhex_bytes_with_sep()
 #ifdef USE_ZLIB_CRC32
-#  include "zlib.h"
+#  include "zlib-ng.h"
 #endif
 
 typedef struct binascii_state {
@@ -776,7 +776,7 @@ binascii_crc32_impl(PyObject *module, Py_buffer *data, unsigned int crc)
         Py_ssize_t len = data->len;
 
         Py_BEGIN_ALLOW_THREADS
-        /* Avoid truncation of length for very large buffers. crc32() takes
+        /* Avoid truncation of length for very large buffers. zng_crc32() takes
            length as an unsigned int, which may be narrower than Py_ssize_t.
            We further limit size due to bugs in Apple's macOS zlib.
            See https://github.com/python/cpython/issues/105967
@@ -786,15 +786,15 @@ binascii_crc32_impl(PyObject *module, Py_buffer *data, unsigned int crc)
 # error "unsupported less than 32-bit platform?"
 #endif
         while ((size_t)len > ZLIB_CRC_CHUNK_SIZE) {
-            crc = crc32(crc, buf, ZLIB_CRC_CHUNK_SIZE);
+            crc = zng_crc32(crc, buf, ZLIB_CRC_CHUNK_SIZE);
             buf += (size_t) ZLIB_CRC_CHUNK_SIZE;
             len -= (size_t) ZLIB_CRC_CHUNK_SIZE;
         }
 #undef ZLIB_CRC_CHUNK_SIZE
-        crc = crc32(crc, buf, (unsigned int)len);
+        crc = zng_crc32(crc, buf, (unsigned int)len);
         Py_END_ALLOW_THREADS
     } else {
-        crc = crc32(crc, data->buf, (unsigned int)data->len);
+        crc = zng_crc32(crc, data->buf, (unsigned int)data->len);
     }
     return crc & 0xffffffff;
 }
diff --git a/Modules/zlibmodule.c b/Modules/zlibmodule.c
index cb360f2..b5260ce 100644
--- a/Modules/zlibmodule.c
+++ b/Modules/zlibmodule.c
@@ -9,7 +9,7 @@
 
 #include "Python.h"
 
-#include "zlib.h"
+#include "zlib-ng.h"
 #include "stdbool.h"
 #include <stddef.h>               // offsetof()
 
-- 
2.51.0

